### PR TITLE
Impliment frontend page for adding movies

### PR DESCRIPTION
**Pull request by Alaskanyoda**
added way for content editor to add movies

### CODE REVIEW

File: frontend/src/views/AddMovieView.vue [hunk 1]
```diff
@@ -1,8 +1,109 @@
...
+  submitSuccess.value = false
+  if (myForm.value === null) {
+    return
+  }
+  const data = JSON.stringify({
+    name: myForm.value.node.value.name,
+    genre: myForm.value.node.value.genre,
+    movieLocation: myForm.value.node.value.movieUrl,
+    comments: ""
+  })
```
Comment (yeSpud) [inline frontend/src/views/AddMovieView.vue:21]:
```suggestion
  error.value = ""
```

File: frontend/src/views/AddMovieView.vue [hunk 2]
```diff
@@ -1,8 +1,109 @@
...
+    const response = await fetch('http://localhost:8080/movies', {
+      credentials: "include",
+      method: "POST",
+      headers: { "Content-Type": "application/json" },
+      body: data
+    })
+
+    if (response.ok) {
+      submitSuccess.value = true
+      error.value = ""
```
Comment (yeSpud) [inline frontend/src/views/AddMovieView.vue:33]:
```suggestion
```

File: frontend/src/views/AddMovieView.vue [hunk 3]
```diff
@@ -1,8 +1,109 @@
...
 
 <template>
-  <main>
-    <h1 style="color: white; text-align: center">Todo: Complete me Ezekiel! :D</h1>
+  <main class="container">
+    <h2 align="center" style="color: white">Add A Movie</h2>
+    <div v-if="error !== ''" class="alert alert-danger" role="alert">Error: {{ error }}</div>
+    <div v-if="submitSuccess" class="alert alert-success" role="alert">Successfully added movie!</div>
+    <FormKit class="form-group" type="form" ref="myForm" @submit="addMovie" :actions="false">
+
```
Comment (yeSpud) [inline frontend/src/views/AddMovieView.vue]:
```suggestion
    <div v-if="submitSuccess" class="alert alert-success" role="alert">
      Successfully added movie!
    </div>
    <FormKit class="form-group" type="form" @submit="addMovie" :actions="false">
```

File: frontend/src/views/AddMovieView.vue [hunk 4]
```diff
@@ -1,8 +1,109 @@
...
 <template>
-  <main>
-    <h1 style="color: white; text-align: center">Todo: Complete me Ezekiel! :D</h1>
+  <main class="container">
+    <h2 align="center" style="color: white">Add A Movie</h2>
+    <div v-if="error !== ''" class="alert alert-danger" role="alert">Error: {{ error }}</div>
+    <div v-if="submitSuccess" class="alert alert-success" role="alert">
+      Successfully added movie!
+    </div>
+    <FormKit class="form-group" type="form" ref="myForm" @submit="addMovie" :actions="false">
```
Comment (yeSpud) [inline frontend/src/views/AddMovieView.vue:57]:
```suggestion
    <FormKit class="form-group" type="form" @submit="addMovie" :actions="false">
```

File: frontend/src/views/AddMovieView.vue [hunk 5]
```diff
@@ -1,8 +1,109 @@
...
+      <FormKit
+        type="text"
+        name="name"
+        label="Movie Name"
+        input-class="form-control"
+        :validation="[['required']]"
+        :validation-messages="{
+          required: 'A movie name is required',
+          characterMax: 'Movie name must be under 200 characters long'
+        }"
```
Comment (yeSpud) [inline frontend/src/views/AddMovieView.vue:67]:
```suggestion
        validation="required"
        :validation-messages="{ required: 'A movie name is required' }"
        }"
```

File: frontend/src/views/AddMovieView.vue [hunk 6]
```diff
@@ -1,8 +1,109 @@
...
+        validation-visibility="live"
+      />
+
+      <FormKit
+        type="select"
+        name="genre"
+        label="Genre(s)"
+        placeholder=" "
```
Comment (yeSpud) [inline frontend/src/views/AddMovieView.vue:75]:
```suggestion
        label="Genre"
        placeholder="Select genre"
```

File: frontend/src/views/AddMovieView.vue [hunk 7]
```diff
@@ -1,8 +1,109 @@
...
+        input-class="form-control"
+        :options="genreList"
+        :validation="[['required']]"
+        :validation-messages="{
+          required: 'A movie name is required',
+          characterMax: 'Movie name must be under 200 characters long'
+        }"
```
Comment (yeSpud) [inline frontend/src/views/AddMovieView.vue:82]:
```suggestion
        validation="required"
        :validation-messages="{ required: 'A movie genre is required' }"
```

File: frontend/src/views/AddMovieView.vue [hunk 8]
```diff
@@ -1,8 +1,109 @@
...
+      />
+
+      <FormKit
+        type="text"
+        name="movieUrl"
+        label="Movie URL"
+        input-class="form-control"
+        :validation="[['required']]"
```
Comment (yeSpud) [inline frontend/src/views/AddMovieView.vue:90]:
```suggestion
        validation="required"
```

File: frontend/src/views/AddMovieView.vue [hunk 9]
```diff
@@ -1,8 +1,109 @@
...
+        }"
+      />
+
+      <div class="row btn-group btn-group-lg" role="group">
+        <FormKit
+          input-class="btn"
+          type="button"
+          label="Cancle"
+          @click="router.push({ name: 'home' })"
+        />
```
Comment (yeSpud) [inline frontend/src/views/AddMovieView.vue:103]:
```suggestion
        <FormKit input-class="btn" type="button" label="Cancle" @click="router.back()" />
```

**Review by yeSpud — CHANGES_REQUESTED** [review]
I notifced that I made an error with reguards to the database giving an error when creating a new movie becasue theres no comment provided.

This should instead be fixed by changing `comments String` to `comments String @default("")` in schema.prisma, and then running `pnpm prisma generate`

File: frontend/src/views/AddMovieView.vue (hunk 9 above)
Comment (Alaskanyoda) [reply frontend/src/views/AddMovieView.vue:102]: the prettier thing changed this, it was originally how u had it

File: frontend/src/views/AddMovieView.vue (hunk 5 above)
Comment (Alaskanyoda) [reply frontend/src/views/AddMovieView.vue:66]: where did the extra closing bracket come from?

File: frontend/src/views/AddMovieView.vue [hunk 10]
```diff
@@ -1,8 +1,108 @@
...
+    genre: myForm.value.node.value.genre,
+    movieLocation: myForm.value.node.value.movieUrl
+  })
+
+  try {
+    const response = await fetch('http://localhost:8080/movies', {
+      credentials: "include",
+      method: "POST",
+      headers: { "Content-Type": "application/json" },
+      body: data
```
[the remark this answers is not in the document]
Comment (Alaskanyoda) [reply frontend/src/views/AddMovieView.vue:27]: how does it know what data goes where?

[the remark this answers is not in the document]
Comment (yeSpud) [reply frontend/src/views/AddMovieView.vue:27]: > how does it know what data goes where?

When the function is called by the submit event of the form it pulls from the child forms. 

So the function has its first argument of the data inside the form (which in our case is name, genre, and movie location)

File: frontend/src/views/AddMovieView.vue (hunk 9 above)
Comment (yeSpud) [reply frontend/src/views/AddMovieView.vue:97]: nO 😭

**Review by yeSpud — APPROVED** [review]
I like what I see 👀

**Pull request merged**